src/components/sidebar/portaled-sidebar.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC, useState, useEffect } from "react"
+import { Portal } from "react-portal"
+import { PortalSidebox } from "./styled"
+
+const ESCAPE_KEY = 27
+
+export const PortalSidebar: FC<PortalSidebarProps> = ({
+  isOpen = true,
+  children,
+  closeOnEsc = false,
+  onClose = () => {},
+  right = false,
+}: PortalSidebarProps) => {
+  const [isOpenState, setOpen] = useState(isOpen)
+
+  useEffect(() => {
+    setOpen(isOpen)
+    document.addEventListener("keydown", (event: KeyboardEvent) => {
+      if (event.keyCode === ESCAPE_KEY && closeOnEsc) {
+        setOpen(false)
+        onClose()
+      }
+    })
+    return () => {
+      document.removeEventListener("keydown", (event: KeyboardEvent) => {
+        if (event.keyCode === ESCAPE_KEY && closeOnEsc) {
+          setOpen(false)
+          onClose()
+        }
+      })
+    }
+  }, [isOpen])
+
+  return isOpenState ? (
```

Comment:
Does it mean we want to have an uncontrolled component? If so, why we still provide isOpen, and why it defaults to true? Can we just assume the panel will be controlled from above, and remove the local state, defaulting etc from inside? 
If we still want this functionality, it's easier to write a wrapper.

Reply:
fixed after discuss